dragonfly_grasshopper/src/DF Building from Footprint.py

```diff
@@ -0,0 +1,121 @@
+# Dragonfly: A Plugin for Environmental Analysis (GPL)
+# This file is part of Dragonfly.
+#
+# Copyright (c) 2019, Ladybug Tools.
+# You should have received a copy of the GNU General Public License
+# along with Dragonfly; If not, see <http://www.gnu.org/licenses/>.
+# 
+# @license GPL-3.0+ <http://spdx.org/licenses/GPL-3.0+>
+
+"""
+Create Dragonfly Buildings from footprint geometry (horizontal Rhino surfaces).
+-
+
+    Args:
+        _footprint_geo: A list of horizontal Rhino surfaces to be converted
+            into Buildings.
+        _floor_to_floor: A list of float values with a length equal to the number of
+            stories in each Building. Each value in the list represents the
+            floor_to_floor height of the Story starting from the first floor and
+            then moving to the top floor.
+        _name_: A base name to be used for the Buildings. This will be combined
+            with the index of each input _footprint_geo to yield a unique name for each
+            output Building.
+        _program_: Text for the program of the Buildings (to be looked up in the
+            ProgramType library) such as that output from the "HB List Programs"
+            component. This can also be a custom ProgramType object. If no program
+            is input here, the Buildings will have a generic office program.
+        _constr_set_: Text for the construction set of the Buildings, which is used
+            to assign all default energy constructions needed to create an energy
+            model. Text should refer to a ConstructionSet within the library such
+            as that output from the "HB List Construction Sets" component. This
+            can also be a custom ConstructionSet object. If nothing is input here,
+            the Buildings will have a generic construction set that is not sensitive
+            to the Buildings's climate or building energy code.
+        conditioned_: Boolean to note whether the Buildings have heating and cooling
+            systems.
+        _run: Set to True to run the component and create Dragonfly Buildings.
+    
+    Returns:
+        report: Reports, errors, warnings, etc.
+        buildings: Dragonfly buildings.
+"""
+
+ghenv.Component.Name = "DF Building from Footprint"
+ghenv.Component.NickName = 'BuildingFootprint'
+ghenv.Component.Message = '0.1.0'
+ghenv.Component.Category = "Dragonfly"
+ghenv.Component.SubCategory = '0 :: Create'
+ghenv.Component.AdditionalHelpFromDocStrings = "2"
+
+# document-wide counter to generate new unique Building names
+import scriptcontext
+try:
+    scriptcontext.sticky["bldg_count"]
+except KeyError:  # first time that the component is running
+    scriptcontext.sticky["bldg_count"] = 1
```

Comment:
If I understand what are you trying to do here correctly you may want to use a combination of component instance guid and component run count. It will give you unique names similar to this approach

![image](https://user-images.githubusercontent.com/2915573/68516336-b6c21280-0251-11ea-9e8c-9f0e9f803161.png)

The advantage is that at some point you don't end up with a large start number for the building after running the script several times.

Reply:
Let me think about this more and let's discuss it over a call.  Using the split uuid might be better than what I have now.

The main reason why I originally wrote it that way was to avoid very long names for buildings, since the names of the Stories, Rooms, Faces, Apertures, and Shades are ultimately derived from this building name by concatenating to it. So there's a risk of hitting the 100 character limit for E+ if default names are too long (like a full uuid4). Just taking the first few characters of the uuid might address this, though.